lib/pf/Portal/Profile.pm

```diff
@@ -415,6 +416,62 @@ sub dot1xRecomputeRoleFromPortal {
     return $self->{'_dot1x_recompute_role_from_portal'};
 }
 
+=item getScans
+
+Returns the Scans IDs for the profile
+
+=cut
+
+sub getScans {
+    my ($self) = @_;
+    return $self->{'_scans'};
+}
+
+=item scanObjects
+
+The scanObjects
+
+=cut
+
+sub scanObjects {
+    my ($self) = @_;
+    return grep { defined $_ } map { pf::factory::scan->new($_) } @{ $self->getScans || [] };
+}
+
+=item findScan
+
+return the first scan that match the device
+
+=cut
+
+sub findScan {
+    my ($self, $mac, $node_attributes) = @_;
+    my $logger = get_logger();
+    my $scancs = pf::ConfigStore::Scan->new();
```

Comment:
We should be using the config hashes instead of the config store because they're faster